test/run-drun/test-cost.mo

```diff
@@ -0,0 +1,38 @@
+import Prim "mo:⛔";
+
+import Cycles "cycles/cycles";
+
+actor client {
+  func print(t : Text) = Prim.debugPrint("client: " # t);
+
+  let ic00 = actor "aaaaa-aa" : actor {
+    create_canister : {
+      settings : ?{
+        controllers : ?[Principal];
+        compute_allocation : ?Nat;
+        memory_allocation : ?Nat;
+        freezing_threshold : ?Nat;
+      };
+    } -> async { canister_id : Principal };
+  };
+
+  public func go() : async () {
+    print(debug_show (Prim.costCreateCanister()) # " -- create canister cost");
+
+    let before0 = Cycles.balance();
+    let before1 = Cycles.balance();
+    let { canister_id } = await ic00.create_canister({ settings = null });
+    let after = Cycles.balance();
+    print("created canister id: " # debug_show (canister_id));
+    print(debug_show(before0) # " -- Cycles.balance() before0");
+    print(debug_show(before1) # " -- Cycles.balance() before1");
+    print(debug_show(after) # " -- Cycles.balance() after");
+    print(debug_show (before1 - after : Nat) # " -- Cycles.balance() diff");
```

Comment:
Output from `--playground`

```
client: 500_000_000_000 -- create canister cost
client: created canister id: 7gngh-jqaaa-aaaab-qacvq-cai
client: 593_554_306_953 -- Cycles.balance() before0
client: 593_554_306_953 -- Cycles.balance() before1
client: 593_548_861_792 -- Cycles.balance() after
client:       5_445_161 -- Cycles.balance() diff

```

Reply:
[500B is the documented cost](https://internetcomputer.org/docs/building-apps/essentials/gas-cost#canister-creation)

Reply:
1. Why the reported canister cost is different than the actual amount?
2. Why `(with cycles = )` was not needed here? Cycles were automatically deducted from the balance.
Whereas on the local replica this call was failing with not enough available cycles.